### PR TITLE
Feature  evaluation pipeline

### PR DESCRIPTION
**Pull request by anty-filidor**
Added minimal working version of the evaluation pipelnie
Moved back implementations of the spreading model
Unified naming across the repository and reorganised scripts

Following issues will be addressed in other, upcoming PRs:
- add repetitive seed set selection (for nondeterministic methods)
- add ground truth seed selector
- add vote rank seed selector
- suppress logging from 3-rd party infmax methods

You can skip everything under `src/icm`, because this code has been moved from another repo and is already after review.

### CODE REVIEW

File: src/evaluators/loader.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+def load_infmax_model(config: dict[str, Any], random_seed: int, nb_seeds: int) -> Any:
+    if config["name"] in {"MultiNode2VecKMeans", "MultiNode2VecKMeansAuto"}:
+        from multi_node2vec_kmeans.loader import load_model
+        if config["parameters"]["rng_seed"] == "auto":
+            config["parameters"]["rng_seed"] = random_seed
+        if config["parameters"]["k_means"]["nb_seeds"] == "auto":
+            config["parameters"]["k_means"]["nb_seeds"] = nb_seeds
+        return load_model({"model": config})
+    else:
+        return lambda x: "dupa"  # TODO: add here GT loader
```
Comment (mateuszStolarski) [inline src/evaluators/loader.py:19]: https://www.youtube.com/watch?v=k8z1hcetWoQ&ab_channel=ProgramujzBeliarem%21 XD

Comment (mateuszStolarski) [reply src/evaluators/loader.py:19]: (Just a meme, nothing to change in code for now)

File: src/evaluators/main_evaluator.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+
+def get_step_func(spreading_model_name: str) -> Callable:
+    if spreading_model_name == nd_model.FixedBudgetMICModel.__name__:
+        raise NotImplementedError(f"Pipeline for {spreading_model_name} is not yet ready!")
+    elif spreading_model_name == torch_model.TorchMICModel.__name__:
+        step_func = step_eval
+    else:
+        raise ValueError(f"Incorrect name of them model {spreading_model_name}")
+    print(f"Inferred step function as: {step_func.__name__}")
```
Comment (mateuszStolarski) [inline src/evaluators/main_evaluator.py:21]: (2) Why we are not using logging here?

**Review by mateuszStolarski — APPROVED** [review]

File: src/evaluators/main_evaluator.py (hunk 2 above)
Comment (anty-filidor) [reply src/evaluators/main_evaluator.py:21]: to be honest - I'm using massively `tqdm` here and if you mix it with `logging` you get neither progress bar nor pretty logs

that's why i decided to stick to `print`s

File: src/evaluators/loader.py (hunk 1 above)
Comment (anty-filidor) [reply src/evaluators/loader.py:19]: :D

File: src/evaluators/main_evaluator.py (hunk 2 above)
Comment (mateuszStolarski) [reply src/evaluators/main_evaluator.py:21]: Ok, may you add todo to check hydra for logging for this case (I'll do it, I just don't want to forget about it)

Comment (anty-filidor) [reply src/evaluators/main_evaluator.py:21]: done

**Pull request merged**